addons/portal/wizard/mail_compose_message.py

```diff
@@ -0,0 +1,28 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+from odoo import api, fields, models
+
+
+class MailComposeMessage(models.TransientModel):
+    _inherit = 'mail.compose.message'
+
+    def default_share_link(self):
+        context = self.env.context
+        active_model = context.get('active_model')
+        active_id = context.get('active_id')
+        model = self.env[active_model]
+        is_model = issubclass(type(model), self.pool['portal.mixin'])
+        if is_model:
+            doc_url = model.browse(active_id).portal_url
+            base_url = self.env['ir.config_parameter'].sudo().get_param('web.base.url')
+            return base_url + doc_url
+        return False
+
+    @api.model
+    def default_get(self, fields):
+        result = super(MailComposeMessage, self).default_get(fields)
+        if self.env.context.get('share_test'):
```

Comment:
Why only do that conditionally? Not doing that would also allow just setting `default=` on the field I think.